### PR TITLE
Fix GitLab repository missing name

### LINKED ISSUE #1643: Missing name of repository in GitLab Docker Registry
I'm using the GitLab registry and I noticed that the name of the current repository from a given registry is missing.

And this behavior can be observed only for GitLab, DockerHub works like a charm.

I'll provide some images so that it's more clear.

Observation:
Image of observed defect

Current state in GitLab:
Current state in GitLab

The same thing, but in DockerHub:
Current state in GitLab

### PR DESCRIPTION
**Pull request by danielgospodinow**
Closes #1643

### CODE REVIEW

File: src/tree/registries/gitLab/GitLabRepositoryTreeItem.ts [hunk 1]
```diff
@@ -17,6 +17,10 @@ export class GitLabRepositoryTreeItem extends RemoteRepositoryTreeItemBase {
     private _nextLink?: string;
 
     public constructor(parent: GitLabProjectTreeItem, id: string, name: string) {
+        if (!name) {
+            name = parent.label;
+        }
+
         super(parent, name);
```
Comment (bwateratmsft) [inline src/tree/registries/gitLab/GitLabRepositoryTreeItem.ts]: An alternative to the `if` block would be this:
`super(parent, name || parent.label)`

This is saying that if `name` is any falsy value (`undefined`, empty string), then use `parent.label`.

File: src/tree/registries/gitLab/GitLabRepositoryTreeItem.ts [hunk 2]
```diff
@@ -17,6 +17,10 @@ export class GitLabRepositoryTreeItem extends RemoteRepositoryTreeItemBase {
     private _nextLink?: string;
 
     public constructor(parent: GitLabProjectTreeItem, id: string, name: string) {
+        if (!name) {
```
Comment (bwateratmsft) [inline src/tree/registries/gitLab/GitLabRepositoryTreeItem.ts]: Can you add a comment as to why? e.g.
`// GitLab returns repositories with empty names, if the project's namespace is used as the repository`

File: src/tree/registries/gitLab/GitLabRepositoryTreeItem.ts (hunk 1 above)
Comment (danielgospodinow) [reply src/tree/registries/gitLab/GitLabRepositoryTreeItem.ts]: Oh, it really looks much better. :+1: I'll refactor this.

**Comment by bwateratmsft** [comment]
I've merged it in, thanks @danielgospodinow!

**Comment by danielgospodinow** [comment]
It was a pleasure, thanks you! :smile:

**Pull request merged**